File: java/src/test/java/com/iteso/nintendo/behaviors/attack/ImpactTruenoAttackTest.java
package com.iteso.nintendo.behaviors.attack;

import com.iteso.nintendo.behaviors.attack.ImpactTruenoAttack;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class ImpactTruenoAttackTest {

    private ImpactTruenoAttack impactTrueno;

    @Before
    public void setUp() {
        impactTrueno =  new ImpactTruenoAttack();
    }

    @Test
    public void attackTest() {
        assertEquals("Attacking opponent with Impactrueno causing a damage of 13",
                     impactTrueno.attack());
    }

        @Test
    public void getAttackNameTest() {
        assertEquals("Impactrueno", impactTrueno.getAttackName());
    }

        @Test
    public void getAttackDamageTest() {
        assertEquals(13, impactTrueno.getAttackDamage());
    }
}

File: python/src/unittest/python/behaviors/attack/ImpactTruenoAttack_test.py
from mockito import mock, verify, when
import unittest
from behaviors.attack.ImpactTruenoAttack import ImpactTruenoAttack

class ImpactTruenoAttack_tests(unnitest.TestCase):

    def setUp(self):
        self.attack_obj = ImpactTruenoAttack()
    
    def test_attack(self):
        self.assertEqual("Attacking opponent with Impactrueno causing a damage of 13",self.attack_obj.attack())

    def test_getAttackName(self):
        self.assertEqual("Impactrueno",self.attack_obj.getAttackName())
   
    def test_getAttackDamage(self):
        self.assertEqual(13,self.attack_obj.getAttackDamage())